fe/fe-core/src/main/java/com/starrocks/catalog/StarOSAgent.java

```diff
@@ -47,15 +59,109 @@ public Set<Long> getBackendIdsByShard(long shardId) {
+        }
+
+        Preconditions.checkState(shardInfos.size() == numShards);
+        return shardInfos.stream().map(shardInfo -> shardInfo.getShardId()).collect(Collectors.toList());
+    }*/
+
+    /*
+    public long getPrimaryBackendIdByShard(long shardId) {
+        Set<Long> backendIds = getBackendIdsByShard(shardId);
+        Preconditions.checkState(backendIds.size() == 1);
+        return backendIds.iterator().next();
+    }*/
+
+    /*
+     public Set<Long> getBackendIdsByShard(long shardId) {
+        Set<Long> backendIds = Sets.newHashSet();
+        List<ShardInfo> shardInfos = Lists.newArrayList();
+        try {
+            shardInfos = client.getShardInfo(1, Lists.newArrayList(shardId));
+        } catch (StarClientException e) {
+            LOG.warn(e);
+            return Sets.newHashSet();
+        }
+
+        Preconditions.checkState(shardInfos.size() == 1);
+        List<ReplicaInfo> replicaInfos = shardInfos.get(0).getReplicaInfoList();
+        for (ReplicaInfo replicaInfo : replicaInfos) {
+            WorkerInfo workerInfo = replicaInfo.getWorkerInfo();
+            String ipPort = workerInfo.getIpPort();
+            String host = ipPort.split(":")[0];
+            long backendId = Catalog.getCurrentSystemInfo().getBackendIdByHost(host);
+            if (backendId == -1) {
+                LOG.warn("Backend does not exists. host: {}", host);
+                continue;
+            }
+            backendIds.add(backendId);
+        }
+        return backendIds;
+    }*/
+
+    /*
+    public long registerAndBootStrapService(String serviceTemplateName, String serviceName) {
+        long serviceId = -1;
+        try {
+            client.registerService(serviceTemplateName);
+        } catch (StarClientException e) {
+            if (e.getCode() != StarClientException.ExceptionCode.ALREADY_EXIST) {
+                LOG.warn(e);
+                return serviceId;
+            }
+        }
+
+        try {
+            serviceId = client.bootstrapService(serviceTemplateName, serviceName);
+            LOG.info("get serviceId: {} from strMgr", serviceId);
+        } catch (StarClientException e) {
+            if (e.getCode() != StarClientException.ExceptionCode.ALREADY_EXIST) {
+                LOG.warn(e);
+            }else{
+                serviceId = client.getService(serviceName);
+            }
+        } finally {
+            return serviceId;
+        }
+    }*/
+
     // Mock StarClient
     private class StarClient {
         // private Map<Long, List<Replica>> shardIdToWorkerIds;
         private Map<Long, Worker> idToWorker;
+        private Map<Integer, Worker> serviceIdToWorker;
+
+        private Set<String> serviceTemplates;
+        private Set<String> services;
 
         private long id = System.currentTimeMillis();
 
         public StarClient() {
             idToWorker = Maps.newHashMap();
+            serviceIdToWorker = Maps.newHashMap();
+            serviceTemplates = Sets.newHashSet();
```

Comment:
serviceTemplates not needed, there will always be one template if FE integrate with staros.